File: Tuan6/app/src/main/java/com/example/test/tuan6/MainActivity.java
package com.example.test.tuan6;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText name;
    private EditText mssv;
    private EditText ns;
    private EditText diachi;
    private EditText email;
    private EditText sdt;
    private RadioGroup gt;
    private CheckBox thethao,dulich,amnhac;
    private Button submit;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        name=findViewById(R.id.hoten);
        mssv =findViewById(R.id.mssv);
        ns=findViewById(R.id.ns);
        diachi=findViewById(R.id.diachi);
        email=findViewById(R.id.mail);
        sdt=findViewById(R.id.phone);
        gt=findViewById(R.id.gt);
        thethao=findViewById(R.id.sttt);
        dulich=findViewById(R.id.stdl);
        amnhac=findViewById(R.id.stan);
        submit=findViewById(R.id.submit);
        submit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                checkInput();
            }
        });
    }
    void checkInput(){
        if (name.getText().toString().isEmpty() || mssv.getText().toString().isEmpty() || ns.getText().toString().isEmpty()
                ||sdt.getText().toString().isEmpty() || email.getText().toString().isEmpty()   ) {
            Toast.makeText(this,"Cần nhập đầy đủ thông tin", Toast.LENGTH_LONG).show();
        }
    }

}

File: Convert/app/src/main/java/com/example/test/convert/MainActivity.kt
package com.example.test.convert

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    val lsHeso= listOf<Int>(2,8,10,16)
    var heso1=0
    var heso2=0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val adapter=ArrayAdapter(this,android.R.layout.simple_spinner_item,lsHeso)
        adapter.setDropDownViewResource(R.layout.support_simple_spinner_dropdown_item)
        from.adapter=adapter
        to.adapter=adapter

        from.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View?,
                position: Int,
                id: Long
            ) {
                heso1=position
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
//                TODO("Not yet implemented")

            }

        }
        to.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
            override fun onItemSelected(
                parent: AdapterView<*>?,
                view: View?,
                position: Int,
                id: Long
            ) {
                heso2=position
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
//                TODO("Not yet implemented")

            }

        }


        reverse.setOnClickListener {
            val t =heso1
            from.setSelection(heso2)
            to.setSelection(heso1)
            heso1=heso2
            heso2=t
        }

        btnConvert.setOnClickListener {
            if(input.text.isNullOrEmpty()){
                Toast.makeText(this,"Nhập giá trị cần chuyển đổi trước",Toast.LENGTH_LONG).show()
            }
            else {
                var res=onConvert(input.text.toString().trim(),lsHeso[heso1],lsHeso[heso2])
                output.text=res
            }
        }
    }
    private fun onConvert(x:String, coso1:Int, coso2:Int):String{
        val t=x.toInt(10)
        val t2=t.toString(coso2)
        return t2
    }
}